go/service/gregor.go

```diff
@@ -690,6 +690,7 @@ func (g *gregorHandler) OnConnect(ctx context.Context, conn *rpc.Connection,
 		if _, ok := err.(libkb.BadSessionError); ok {
 			g.chatLog.Debug(ctx, "bad session from SyncAll(): forcing session check on next attempt")
 			g.forceSessionCheck = true
+			nist.DidFail()
```

Comment:
This did fix a bug, but only the bug that reconnect always failed after it failed once.  It doesn't fix the original failure.